### PR TITLE
Cromwell restart

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/bin/travis/resources/centaur.wdl [hunk 1]
```diff
@@ -21,7 +33,7 @@ task centaur {
     }
 
     runtime {
-        docker: "geoffjentry/centaur-cromwell:latest" 
+        docker: "us.gcr.io/broad-dsde-cromwell-dev/centaur:latest"
```
Comment (mcovarr) [inline src/bin/travis/resources/centaur.wdl]: :1st_place_medal:

File: src/bin/travis/resources/centaur.wdl [hunk 2]
```diff
@@ -6,6 +6,18 @@ task centaur {
     File token
 
     command<<<
+        # start mysql server
+        mysqld &
+        
+        # give time to server to start
+        sleep 10
```
Comment (mcovarr) [inline src/bin/travis/resources/centaur.wdl]: hmmm

File: engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaCheckingCacheEntryExistenceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      createCheckingCacheEntryExistenceEjea()
+      
+      ejea ! HasCallCacheEntry(CallCacheEntryForCall(helper.workflowId, helper.jobDescriptorKey))
+      helper.jobPreparationProbe.expectMsg(awaitTimeout, "expecting CallPreparation Start", CallPreparation.Start)
+      ejea.stateName should be(PreparingJob)
+      
+      ejea.underlyingActor.effectiveCallCachingMode shouldBe CallCachingOff
+    }
+
+    "prepare job if a cache entry already exists for this job" in {
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaCheckingCacheEntryExistenceSpec.scala]: should that say "if not cache entry already exists"?

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala [hunk 4]
```diff
@@ -121,22 +121,44 @@ class EngineJobExecutionActor(replyTo: ActorRef,
...
       throw new RuntimeException(f.reason)
   }
 
+  // When we're restarting but the job store says the job is not complete.
+  // This is to cover for the case where Cromwell was stopped after writing Cache Info to the DB but before
+  // writing to the JobStore. In that case, we don't want to cache to ourselves (turn cache read off), nor do we want to 
+  // try and write the cache info again, which would fail (turn cache write off).
+  // This means call caching should be disabled.
+  // Note that we check that there is not a Cache entry for *this* current job. It's still technically possible
+  // to call cache to another job that finished while this one was running (before the restart).
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala]: nice description!

**Comment by mcovarr** [comment]
:+1:

File: src/bin/travis/resources/centaur.wdl [hunk 5]
```diff
@@ -6,6 +6,26 @@ task centaur {
     File token
 
     command<<<
+        # start mysql server
+        mysqld &
+        
+        # give time to server to start
+        git clone https://github.com/vishnubob/wait-for-it.git
+        cd wait-for-it
+        ./wait-for-it.sh -t 10 localhost:3306
```
Comment (mcovarr) [inline src/bin/travis/resources/centaur.wdl]: Are we really sure this doesn't mine Bitcoins or do anything nastier while it's supposedly waiting?  🙂

Comment (Horneth) [inline src/bin/travis/resources/centaur.wdl]: Yes :) It was recommended by bernick when I did the docker compose with mysql script thing

Comment (geoffjentry) [inline src/bin/travis/resources/centaur.wdl]: How do we know he's not bitcoin mining? :)

File: src/bin/travis/resources/centaur.wdl (hunk 1 above)
Comment (geoffjentry) [inline src/bin/travis/resources/centaur.wdl]: :')

(kidding)

**Comment by geoffjentry** [comment]
👍

File: src/bin/travis/resources/centaur.wdl (hunk 1 above)
Comment (mcovarr) [inline src/bin/travis/resources/centaur.wdl]: I think you've done at least a significant piece of [this ticket](https://github.com/broadinstitute/cromwell/issues/1296)

Comment (geoffjentry) [inline src/bin/travis/resources/centaur.wdl]: @mcovarr sadly no. That's referring to all the encrypted secret stuff which sits on my laptop, which I currently don't have access to :)

Comment (mcovarr) [inline src/bin/travis/resources/centaur.wdl]: 😭

**Pull request merged**